controllers/apartments.rb

```diff
@@ -1,14 +1,42 @@
 # index
 get '/apartments' do
-  erb :"apartments/index"
+  @apartments = Apartments.all
+  erb(:"apartments/index")
 end
 
 # new
 get '/apartments/new' do
-  erb :"apartments/new"
+  erb(:"apartments/new")
+end
+
+#create 
+post '/apartments' do
+  @apartment = Apartment.create!(params[:apartment])
+  redirect("/apartments/#{@apartment.id}")
 end
 
 # show
 get '/apartments/:id' do
-  erb :"apartments/show"
+  @apartment = Apartment.find(params[:id])
+  erb(:"apartments/show")
+end
+
+#edit
+get "/apartment/:id/edit" do
+  @apartment = Apartment.find(params[:id])
+  erb(:"apartments/edit")
+end
+
+#update
+put "/apartments/:id" do
+    @apartment = Apartment.find(params[:id])
+    @apartment.update(params[:apartment])
```

Comment:
Suggest using `@apartment.update!(params[:apartment]), so that it raises an error if there is a problem saving.